hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -276,10 +279,33 @@ public Integer apply(HazelcastProperties properties) {
      * <p>
      * The default is depends on the number of available processors. If the available processors count is
      * smaller than 20, there will be 3+3 io threads, otherwise 4+4.
+     *
+     * If SSL is enabled, then the default number of IO threads will be corecount/2.
      */
     public static final HazelcastProperty IO_THREAD_COUNT
-            = new HazelcastProperty("hazelcast.io.thread.count",
-            properties -> Runtime.getRuntime().availableProcessors() >= 20 ? 4 : 3);
+            = new HazelcastProperty("hazelcast.io.thread.count", new Function<HazelcastProperties, Integer>() {
+        @Override
+        public Integer apply(HazelcastProperties properties) {
+            Config config = properties.getConfig();
+            if (config == null) {
+                return getDefault();
+            }
+            NetworkConfig networkConfig = config.getNetworkConfig();
+            if (networkConfig == null) {
+                return getDefault();
+            }
+            SSLConfig sslConfig = networkConfig.getSSLConfig();
+            if (sslConfig == null || !sslConfig.isEnabled()) {
+                return getDefault();
+            }
+            return Math.max(getDefault(), RuntimeAvailableProcessors.get() / 2);
```

Comment:
This could be optimized to avoid double `RuntimeAvailableProcessors.get()` call.